lib/engine/auto_router.rb

```diff
@@ -88,81 +133,328 @@ def compute(corporation, **opts)
           id = chains.flat_map { |c| c[:paths] }.sort!
           next if connections[id]
 
-          connections[id] = chains.map do |c|
+          connection = chains.map do |c|
             { left: c[:nodes][0], right: c[:nodes][1], chain: c }
           end
-        end
-      end
+          connections[id] = connection
 
-      puts "Found #{connections.size} paths in: #{Time.now - now}"
-      puts 'Pruning paths to legal routes'
+          counter += 1
 
-      now = Time.now
-      train_routes = Hash.new { |h, k| h[k] = [] }
-      connections.each do |_, connection|
-        @game.route_trains(corporation).each do |train|
-          route = Engine::Route.new(
-            @game,
-            @game.phase,
-            train,
-            connection_data: connection,
-          )
-          route.revenue
-          train_routes[train] << route
-        rescue GameError # rubocop:disable Lint/SuppressedException
+          # each train has opportunity to vote to abort a branch of this node's path-walk tree
+          path_abort.each { |train, _| path_abort[train] = false }
```

Comment:
looks like another unecessary line

Reply:
this is unaddressed

Reply:
but up ahead you do  path_abort = Hash.new { |h, k| h[k] = false }   which makes each key default false

Reply:
Yeah the code was sloppy.  I've tried to clarify it.  Up top, declare and populate a hash of train to <something>, only setting keys (but having to provide initial values).  In this loop here, set all trains' path_abort to false, so only if they all raise RouteTooLong on the connection will it abort the path.  Path_aborts are cleared for every connection.  And remove the unless path_abort[train], since that was dead code (all flags are falsed right before the loop)

Previously, the code only declared the shape of the hash up top, and here on the first connection loop the path_abort had no entires, but each "unless path_abort[train]" caused the creation of the train,false kvp.  Subsequent connection loops would proceed with all kvp's in place.  That's why I call that sloppy.

Note: I had tried to make path_abort longer-lived as the connections build, but had some logic problem that I never figured out.  The unless is actually unneeded for now, given that they're all falsed before the trains.each,; I had been keeping it here in case I get back to path_abort surviving path.walk yields.  (E.g a 5 train will abort on a 6 route, but a 2nd 10 train will keep accumulating 7,8,9,10,11 routes while the path_abort[5] would prevent 5-route.revenue on 7,8,9,10,11 routes)